agent/api/container/port_binding.go

```diff
@@ -30,7 +32,9 @@ const (
 // PortBinding represents a port binding for a container
 type PortBinding struct {
 	// ContainerPort is the port inside the container
-	ContainerPort uint16
+	ContainerPort *uint16
+	// ContainerPortRange is a range of ports exposed inside the container
+	ContainerPortRange *string
```

Comment:
Since Port 0 and PortRange "" are not valid values, we could have used those as nil checks. There doesn't seem to be a need to update all fields and code use references.